ksqldb-common/src/main/java/io/confluent/ksql/util/StreamsTaskMetadata.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.streams.processor.TaskMetadata;
+
+public class StreamsTaskMetadata {
+
+  private final String taskId;
+  private final Set<TopicPartition> topicPartitions;
+  private final ImmutableMap<TopicPartition, Long> endOffsets;
```

Comment:
I don't think `TopicPartition` is the model we use in our API. To make sure things stay compatible and that we don't add unnecessary fields, convert these to `TopicPartitionEntity`

Reply:
@rodesai I converted these, however there is a key deserializer. I am adding a  `TopicPartitionEntityDeserializer extends KeyDeserializer` to `TopicPartitionEntity`. Or is there some other way this is normally handled?